### DISCUSSION TITLE
Yelp API was working but can no longer get a response on local

### DISCUSSION THREAD

**Discussion by lindseysatterfield** [nss-evening-cohort-14/evening-client-side#324] [Capstones!]
#### Searched other tickets?
Yes, #297 mentioned CORS which I am using, but it did not help my issue.

#### Describe your issue
After finally figuring out the right call to start console warning data back from the Yelp API on my local, I was getting back exactly what I wanted in the console. Five businesses in Nashville for whatever type of cuisine I was looking for. I stopped for the day after getting that right. I am now working on it again and instead of getting data back from the Yelp API in my console, I'm getting a `403 forbidden` error. I just don't understand why it would work yesterday but not today. I wonder if I was getting data back yesterday because I clicked this CORS demo server (https://cors-anywhere.herokuapp.com/corsdemo). Tried that again today and then it throws me a `401 unauthorized` error. I'm wondering if this is a CORS issue after reading some of the sites I've linked down below.

#### List any error messages that appear
`GET https://cors-anywhere.herokuapp.com/https://api.yelp.com/v3/businesses/search?term=american&location=nashville&limit=5 403 (Forbidden)`
and
`Uncaught (in promise) Error: Request failed with status code 403`
I have also gotten at some point: 
`GET https://cors-anywhere.herokuapp.com/https://api.yelp.com/v3/businesses/search?term=american&location=nashville&limit=5 401 (Unauthorized)`
and
`Uncaught (in promise) Error: Request failed with status code 401`

#### What have you googled?
https://restfulapi.net/http-status-codes/#:~:text=403%20(Forbidden),401%20(%E2%80%9CUnauthorized%E2%80%9D).
https://discuss.codecademy.com/t/ravenous-project-part-4-error-403-forbidden/574098
https://discuss.codecademy.com/t/cors-anywhere-issue-with-ravenous-project/569289/12
https://github.com/Rob--W/cors-anywhere/issues/301

#### What have you tried? Tell us everything you have tried.
- Looking over my code to see if I accidentally changed something from when it was working, but I have not changed anything
- Refreshing my API key and using the new one
- Tried changing the CORS url after looking at some of the sites above

#### Link to code
https://github.com/lindseysatterfield/nashville-night-out/blob/yelp/src/helpers/Yelp/YelpData.js
https://github.com/lindseysatterfield/nashville-night-out/blob/yelp/src/helpers/apiKeys.js
https://github.com/lindseysatterfield/nashville-night-out/blob/yelp/src/helpers/views/Search.js

**Comment by lindseysatterfield** [comment]
UPDATE - after restarting my server a few times, I am now getting a response from the Yelp server with the data I need. But I'm afraid it is because I am using that CORS demo that I linked above. Unsure of how to proceed if this only works when using the demo if that's the case.

**Reply by TrinityChristiana** [reply]
Using the CORS Demo is not sustainable. To use it, you have to click a button allowing temporary access. So to get this working, you'll need to find another way to make the API Call. 

Unfortunately, because this is not a part of your MVP, we're not going to prioritize this ticket above other people that do need help completing their MVP. And I'm not sure if we will be able to help you in time for the Soft Demo on Saturday. But I will say you'll have to find another way to make the API call without the CORS Demo. So I'd start looking for other solutions.
